### PR TITLE
MGDOBR-1023 Upload performance test results to Horreum

### PR DESCRIPTION
**Pull request by sutaakar**
[MGDOBR-1023](https://issues.redhat.com/browse/MGDOBR-1023)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by sutaakar** [comment]
Waiting for https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1031 to be merged first, then the PR will be ready for review.

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/HorreumResource.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import software.tnb.horreum.service.Horreum;
+
+public class HorreumResource {
+    private static final boolean HORREUM_RESULTS_UPLOAD = Boolean.valueOf(System.getProperty("performance.horreum.results.upload"));
+    private static final String HORREUM_TEST_RESULT_SCOPE = "PUBLIC";
+
+    private static String HORREUM_TEAM_NAME;
+    private static String HORREUM_TEST_SCHEMA;
+
+    public static Horreum horreum = ServiceFactory.create(Horreum.class);
```
Comment (afalhambra) [inline integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/HorreumResource.java:20]: would be nice to update the sample file we have for the [localconfig.yaml](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/integration-tests/localconfig-example.yaml) file regarding TNB config, so it can be pretty straight forward to test locally

File: integration-tests/src/test/resources/performance-features/send-cloud-events.feature [hunk 2]
```diff
@@ -65,7 +65,7 @@ Feature: Sending cloud events performance tests
           increasingRate:
             initialUsersPerSec: 1
             targetUsersPerSec: 10
-            maxSessions: 10
+            maxSessions: 20
```
Comment (jiripetrlik) [inline integration-tests/src/test/resources/performance-features/send-cloud-events.feature:68]: Is this related to upload to Horreum?

**Review by jiripetrlik — APPROVED** [review]
Looks good to me. I have only one minor question.

File: integration-tests/src/test/resources/performance-features/send-cloud-events.feature (hunk 2 above)
Comment (sutaakar) [reply integration-tests/src/test/resources/performance-features/send-cloud-events.feature:68]: No, I have just noticed that some tests executed locally crashed on lack of sessions, so adding it here.

File: integration-tests/pom.xml [hunk 3]
```diff
@@ -241,6 +257,9 @@
                 <undefined.placeholder.check.enabled>false</undefined.placeholder.check.enabled> <!-- Disable the check as Hyperfoil placeholders use same format -->
                 <performance.webhook.url></performance.webhook.url>
                 <performance.hyperfoil.url></performance.hyperfoil.url>
+                <performance.horreum.team.name>rhose-team</performance.horreum.team.name>
+                <performance.horreum.test.schema>http://hyperfoil.io/run-schema/v3.0</performance.horreum.test.schema>
+                <performance.horreum.results.upload>false</performance.horreum.results.upload>
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply integration-tests/pom.xml:262]: Sure, I will put it in `properties` section of this pom.

File: integration-tests/src/test/resources/performance-features/create-bridges-at-once.feature [hunk 4]
```diff
@@ -103,6 +103,7 @@ Feature: Create Bridges performance tests
               - restartSequence
       """
     Then run benchmark "rhose-create-bridges" within 15 minutes
+    And the benchmark run "rhose-create-bridges" was executed successfully
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply integration-tests/src/test/resources/performance-features/create-bridges-at-once.feature:106]: That is a good question.
Initially the scenario was executed to make sure that RHOSE can handle multiple Bridge creations. The verification is done as part of the tests, so there are no performance results, just passed/failed.
I am thinking about covering also Bridge creation stats (average Bridge creation time and such), for that I will have to adjust the test.

File: integration-tests/pom.xml [hunk 5]
```diff
@@ -19,6 +19,17 @@
...
+  <!-- Using custom version of okhttp due to dependency misalignment between RHOSE and TNB. -->
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.squareup.okhttp3</groupId>
+        <artifactId>okhttp</artifactId>
+        <version>4.9.2</version>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply integration-tests/pom.xml:31]: The version set here is used by TNB. Without this section an older version of `okhttp` is used which lack of some functionality needed by Horeum TNB addon.

**Comment by sutaakar** [comment]
@afalhambra Can you please check again?

File: integration-tests/pom.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (afalhambra) [reply integration-tests/pom.xml:37]: understood, thanks

**Review by afalhambra — APPROVED** [review]
Great work, thanks @sutaakar

**Comment by sutaakar** [comment]
Test failures unrelated, merging.
